internal/filewatcher.go

```diff
@@ -0,0 +1,106 @@
+package internal
+
+import (
+	"fmt"
+	"path/filepath"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"golang.org/x/exp/slog"
```

Comment:
Probably mean (will need to format imports)
```suggestion
	"log/slog"
```